src/Computation/Values/And.cs

```diff
@@ -34,17 +34,18 @@ private static IValue Create(IValue left, ConstantUnsigned right)
         return left switch
         {
             IConstantValue l => l.AsUnsigned().And(right),
-            And l => new And(l._left, Create(l._right, right)),
+            And l => Create(l._left, Create(l._right, right)),
```

Comment:
This was the problem. If `Create(l._right, right)` hits the `IConstantValue l` case and the result of that `And` on the two unsigned values returns `0` then we should be able to short circuit again, so by calling `Create` we put this new value back through the grinder.

The specific example I saw was with two constants of `1` and `244`.